server/services/store/sqlstore/data_migrations.go

```diff
@@ -547,3 +549,91 @@ func (s *SQLStore) getBoardUserTeams(tx sq.BaseRunner, board *model.Board) (map[
 
 	return userTeams, nil
 }
+
+func (s *SQLStore) runDeletedMembershipBoardsMigration() error {
+	if !s.isPlugin {
+		return nil
+	}
+
+	setting, err := s.GetSystemSetting(DeletedMembershipBoardsMigrationKey)
+	if err != nil {
+		return fmt.Errorf("cannot get deleted membership boards migration state: %w", err)
+	}
+
+	// If the migration is already completed, do not run it again.
```

Comment:
Is this a migration that would be safe and useful to run every time?

Reply:
That's a very good point, but I don't think we should run this migration every time, as with the permissions system in place, there are "legal" ways of getting to that point. What we should do probably is to spec out what happens when a user leaves a team / is completely deleted, and that user is the only admin of a board.

We could probably mimic the same thing that channels is doing for channel admins right now